www/js/controller.js

```diff
@@ -594,16 +594,27 @@ define([
 
 			if(model.getNodeIDFor){
 				var d = registry.byId(this.controlMap.description);
-				array.forEach(this._model.authored.getDescriptions(), function(desc){
+				var v = registry.byId(this.controlMap.variable);
+				array.forEach(this._model.authored.getDescriptions(), lang.hitch(this, function(desc){
 					var exists =  model.getNodeIDFor(desc.value);
-					
+
 					if( d.getOptions !== undefined && d.getOptions(desc)) {
 						d.getOptions(desc).disabled=exists;
 						if(desc.value == nodeName){
 							d.getOptions(desc).disabled=false;
 						}
 					}
-				});
+					var variable_name = this._model.authored.getName(desc.value);
+					var option = {label:variable_name, value: variable_name};
+					if(v.getOptions(option)){
+						var disabled = false;
+						if(exists!== null && model.getName(exists)!== undefined)
+							disabled = true
+						v.getOptions(option).disabled=disabled;
+					}
+					// To reflect the changes to UI
+					v.startup();
```

Comment:
add the logic to disable the list in model.js checking which are nodes exist. If not then we should look at why it wasnt working earlier? I dont understand it seems that you are doing the whole logic again. Same logic that has been done just above it.